src/lib/util/SfdxUtils.ts

```diff
@@ -0,0 +1,48 @@
+import childProcess = require('child_process');
+
+/**
+ * A variable to store SFDX's version number, so we don't have to keep running `sfdx -v`.
+ */
+let SFDX_VERSION: string;
+
+/**
+ * Returns the current version of SFDX installed on the machine, or `unknown` if the version cannot be determined.
+ */
+export async function getSfdxVersion(): Promise<string> {
+	// If we already have a cached value, we can just return that instead of doing everything else.
+	if (SFDX_VERSION !== undefined) {
+		return SFDX_VERSION;
+	}
+
+	// Get the output of `sfdx -v`.
+	let rawVersionString: string;
+	try {
+		rawVersionString = await new Promise<string>((res, rej) => {
+			childProcess.exec('sfdx -v', (error, stdout, stderr) => {
```

Comment:
Assuming the user installed `sfdx-cli` using node at all, they did so globally. That means we can't get the version through a simple `require()` call, which is why we do this weirdness.